package/samplers/smac_sampler/sampler.py

```diff
@@ -0,0 +1,274 @@
+                scenario=scenario,
+                n_configs=n_configs,
+                n_configs_per_hyperparameter=n_configs_per_hyperparameter,
+                max_ratio=max_ratio,
+            )
+        elif init_design_type == "lhd":
+            init_design = LatinHypercubeInitialDesign(
+                scenario=scenario,
+                n_configs=n_configs,
+                n_configs_per_hyperparameter=n_configs_per_hyperparameter,
+                max_ratio=max_ratio,
+            )
+        elif init_design_type == "random":
+            init_design = RandomInitialDesign(
+                scenario=scenario,
+                n_configs=n_configs,
+                n_configs_per_hyperparameter=n_configs_per_hyperparameter,
+                max_ratio=max_ratio,
+            )
+        else:
+            raise NotImplementedError(f"Unknown Initial Design Type: {init_design_type}")
+        return init_design
+
+    def sample_relative(
+        self, study: Study, trial: FrozenTrial, search_space: dict[str, BaseDistribution]
+    ) -> dict[str, float]:
+        trial_info: TrialInfo = self.smac.ask()
+        cfg = trial_info.config
+        self._runs_instance_seed_keys[get_config_hash(cfg)] = (
+            trial_info.instance,
+            trial_info.seed,
+        )
+        params = {}
+        for name, hp_value in cfg.items():
+            if name in self._hp_scale_value:
+                hp_value = self._integer_to_step_hp(
+                    integer_value=hp_value, scale_info=self._hp_scale_value[name]
+                )
+            params[name] = hp_value
+
+        return params
+
+    def after_trial(
+        self,
+        study: Study,
+        trial: FrozenTrial,
+        state: TrialState,
+        values: Sequence[float] | None,
+    ) -> None:
+        # Transform the trail info to smac
+        params = trial.params
+        cfg_params = {}
+        for name, hp_value in params.items():
+            if name in self._hp_scale_value:
+                hp_value = self._step_hp_to_intger(hp_value, scale_info=self._hp_scale_value[name])
+            cfg_params[name] = hp_value
+
+        # params to smac HP
+        y = np.asarray(values)
+        if state == TrialState.COMPLETE:
+            status = StatusType.SUCCESS
+        elif state == TrialState.RUNNING:
+            status = StatusType.RUNNING
+        else:
+            status = StatusType.CRASHED
+        trial_value = TrialValue(y, status=status)
+
+        cfg = Configuration(configuration_space=self._cs, values=cfg_params)
+        # Since Optuna does not provide us the
+        instance, seed = self._runs_instance_seed_keys[get_config_hash(cfg)]
+        info = TrialInfo(cfg, seed=seed, instance=instance)
+        self.smac.tell(info=info, value=trial_value, save=False)
+
+    def _transform_step_hp_to_integer(
+        self, distribution: FloatDistribution | IntDistribution
+    ) -> tuple[int, tuple[int | float]]:
```

Comment:
Did you probably mean this?
```suggestion
    ) -> tuple[int, tuple[int | float, int | float]]]:
```